Test/PublishPowerShellModule.Tests.ps1

```diff
@@ -17,15 +17,13 @@ function GivenCredential
 {
     param(
         [Parameter(Mandatory)]
-        [pscredential]
-        $Credential,
+        [pscredential]$Credential,
 
         [Parameter(Mandatory)]
-        [string]
-        $WithID
+        [string]$WithID
     )
 
-    $credentials[$WithID] = $Credential
+    $script:credentials[$WithID] = $Credential

```

Comment:
Since you're setting the contents of the hashtable, you don't need `script:`.

You only need the `script:` scope when changing the value of a script-level variable. PowerShell sets variables locally even if there is an equivalent global variable.